src/js/Form.js

```diff
@@ -1986,9 +2004,11 @@ define( function( require, exports, module ) {
                 if ( eventType === 'validate' ) {
                     that.setInvalid( $input, 'required' );
                 }
+
+                valid = Promise.resolve( false );
```

Comment:
This fixes a (potential) bug in the Promise validation code - if the only validation error was a `required` field, it is otherwise ignored.